### PR TITLE
#167 fontify triple-quoted-string, fixed

### PR DESCRIPTION
**Pull request by andreas-roehler**

### CODE REVIEW

File: elixir-mode.el [hunk 1]
```diff
@@ -273,6 +254,99 @@ for the Elixir programming language."
...
+    (goto-char start)
+    (funcall
+     (syntax-propertize-rules
+      ((elixir-rx string-delimiter)
+       (0 (ignore (elixir-syntax-stringify))))
+      ((rx (group "#{" (0+ (not (any "}"))) "}"))
+       (0 (ignore (elixir-syntax-propertize-interpolation)))))
+     start end)))
+
+;; (defun elixir-syntax-propertize-function (start end)
```
Comment (mattdeboard) [inline elixir-mode.el]: Please delete lines 327-335

**Comment by mattdeboard** [comment]
This looks very good to me! I think automatic indentation would be really valuable but that can definitely wait for a different PR. 

Once those lines are deleted I will merge this in. This will get released ASAP.

File: elixir-mode.el (hunk 1 above)
Comment (andreas-roehler) [inline elixir-mode.el]: Am 26.05.2015 um 20:49 schrieb Matt DeBoard:

> Please delete lines 327-335
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/elixir-lang/emacs-elixir/pull/203/files#r31066466>.
>
This is obsolet now. There is a new pull-request. Fixes TQS only.

**Comment by tonini** [comment]
@mattdeboard @andreas-roehler this is awesome!  :shipit:

**Comment by mattdeboard** [comment]
I have a little file called `foo.ex` I use to manually poke at things like this.

I added a `@heredoc` to a fn like so to test the capability, and it all looks great.

**Comment by mattdeboard** [comment]
```elixir
  @heredoc"""
  How "is" everyone r{doing} today, %Foo{hmm}??"
  """
  def foobar( %ExUnit.Test{} = test ) do
    a_case = test.case
                      stuff()
  end
```

All the various special cases and stuff just retain their string highlighting which is perfect.

**Comment by tonini** [comment]
This is amazing! :) We need to release this as you said ASAP :D

File: elixir-mode.el (hunk 1 above)
Comment (mattdeboard) [inline elixir-mode.el]: Not sure I understand? This is in the new PR (#203)

Comment (andreas-roehler) [inline elixir-mode.el]: Am 26.05.2015 um 20:59 schrieb Matt DeBoard:
>
> In elixir-mode.el 
> <https://github.com/elixir-lang/emacs-elixir/pull/203#discussion_r31067476>:
>
> > +    (0 (ignore (elixir-syntax-stringify))))
> > +   ((rx (group "#{" (0+ (not (any "}"))) "}"))
> > +    (0 (ignore (elixir-syntax-propertize-interpolation))))))
> > +
> > +(defun elixir-syntax-propertize-function (start end)
> > +  (let ((case-fold-search nil))
> > +    (goto-char start)
> > +    (funcall
> > +     (syntax-propertize-rules
> > +      ((elixir-rx string-delimiter)
> > +       (0 (ignore (elixir-syntax-stringify))))
> > +      ((rx (group "#{" (0+ (not (any "}"))) "}"))
> > +       (0 (ignore (elixir-syntax-propertize-interpolation)))))
> > +     start end)))
> > +
> > +;; (defun elixir-syntax-propertize-function (start end)
>
> Not sure I understand? This is in the new PR (#203 
> <https://github.com/elixir-lang/emacs-elixir/pull/203>)
>

Yes. Could you precise your question?

First part of syntax-propertize-rules is the fix.
Second part non-functional for the moment (interpolation is broken as 
announced)

Will go on here after merge.

Comment (mattdeboard) [inline elixir-mode.el]: No worries, you have done enough work for the project, I'll take care of these little cleanups. Thank you so much, this has been a long time coming!

Comment (mattdeboard) [inline elixir-mode.el]: All I meant was
https://github.com/elixir-lang/emacs-elixir/commit/bc2aa4f4f8fd4c96d2d8247b50d5568a9aff7a3e
... but no worries was simple enough to do myself. Thanks again I am sure
tons of users will be very happy at your improvements.

Comment (andreas-roehler) [inline elixir-mode.el]: Am 26.05.2015 um 21:30 schrieb Matt DeBoard:
>
> In elixir-mode.el 
> <https://github.com/elixir-lang/emacs-elixir/pull/203#discussion_r31070502>:
>
> > +    (0 (ignore (elixir-syntax-stringify))))
> > +   ((rx (group "#{" (0+ (not (any "}"))) "}"))
> > +    (0 (ignore (elixir-syntax-propertize-interpolation))))))
> > +
> > +(defun elixir-syntax-propertize-function (start end)
> > +  (let ((case-fold-search nil))
> > +    (goto-char start)
> > +    (funcall
> > +     (syntax-propertize-rules
> > +      ((elixir-rx string-delimiter)
> > +       (0 (ignore (elixir-syntax-stringify))))
> > +      ((rx (group "#{" (0+ (not (any "}"))) "}"))
> > +       (0 (ignore (elixir-syntax-propertize-interpolation)))))
> > +     start end)))
> > +
> > +;; (defun elixir-syntax-propertize-function (start end)
> All I meant was 
> https://github.com/elixir-lang/emacs-elixir/commit/bc2aa4f4f8fd4c96d2d8247b50d5568a9aff7a3e 
> ... but no worries was simple enough to do myself. Thanks again I am 
> sure tons of users will be very happy at your improvements.
> … <#>
> On Tue, May 26, 2015 at 2:11 PM, Andreas Röhler 
> <notifications@github.com> wrote: In elixir-mode.el 
> <https://github.com/elixir-lang/emacs-elixir/pull/203#discussion_r31068707> 
> : > + (0 (ignore (elixir-syntax-stringify)))) > + ((rx (group "#{" (0+ 
> (not (any "}"))) "}")) > + (0 (ignore 
> (elixir-syntax-propertize-interpolation)))))) > + > +(defun 
> elixir-syntax-propertize-function (start end) > + (let 
> ((case-fold-search nil)) > + (goto-char start) > + (funcall > + 
> (syntax-propertize-rules > + ((elixir-rx string-delimiter) > + (0 
> (ignore (elixir-syntax-stringify)))) > + ((rx (group "#{" (0+ (not 
> (any "}"))) "}")) > + (0 (ignore 
> (elixir-syntax-propertize-interpolation))))) > + start end))) > + > 
> +;; (defun elixir-syntax-propertize-function (start end) Am 26.05.2015 
> um 20:59 schrieb Matt DeBoard: In elixir-mode.el < 
> https://github.com/elixir-lang/emacs-elixir/pull/203#discussion_r31067476>: 
> > + (0 (ignore (elixir-syntax-stringify)))) > + ((rx (group "#{" (0+ 
> (not (any "}"))) "}")) > + (0 (ignore 
> (elixir-syntax-propertize-interpolation)))))) > + > +(defun 
> elixir-syntax-propertize-function (start end) > + (let 
> ((case-fold-search nil)) > + (goto-char start) > + (funcall > + 
> (syntax-propertize-rules > + ((elixir-rx string-delimiter) > + (0 
> (ignore (elixir-syntax-stringify)))) > + ((rx (group "#{" (0+ (not 
> (any "}"))) "}")) > + (0 (ignore 
> (elixir-syntax-propertize-interpolation))))) > + start end))) > + > 
> +;; (defun elixir-syntax-propertize-function (start end) Not sure I 
> understand? This is in the new PR (#203 
> <https://github.com/elixir-lang/emacs-elixir/pull/203> 
> <https://github.com/elixir-lang/emacs-elixir/pull/203> < 
> https://github.com/elixir-lang/emacs-elixir/pull/203> 
> <https://github.com/elixir-lang/emacs-elixir/pull/203%3E>) Yes. Could 
> you precise your question? First part of syntax-propertize-rules is 
> the fix. Second part non-functional for the moment (interpolation is 
> broken as announced) Will go on here after merge. — Reply to this 
> email directly or view it on GitHub 
> <https://github.com/elixir-lang/emacs-elixir/pull/203/files#r31068707>.
>
> —
>

Well, glad being useful. Decently translated some work done at 
python-mode. Should mention Dave Love, who was the first solving this AFAIK.

**Pull request merged**